test/Microsoft.AspNetCore.Sockets.Tests/HttpConnectionDispatcherTests.cs

```diff
@@ -77,31 +44,20 @@ public class HttpConnectionDispatcherTests
         [InlineData("/poll")]
         public async Task EndpointsThatRequireConnectionId400WhenNoConnectionIdProvided(string path)
         {
-            var manager = new ConnectionManager();
-            var dispatcher = new HttpConnectionDispatcher(manager, new LoggerFactory());
-            using (var strm = new MemoryStream())
-            {
-                var context = new DefaultHttpContext();
-                context.Response.Body = strm;
-                var services = new ServiceCollection();
-                services.AddSingleton<TestEndPoint>();
-                context.RequestServices = services.BuildServiceProvider();
-                context.Request.Path = path;
-
-                await dispatcher.ExecuteAsync<TestEndPoint>("", context);
-
-                Assert.Equal(StatusCodes.Status400BadRequest, context.Response.StatusCode);
-                await strm.FlushAsync();
-                Assert.Equal("Connection ID required", Encoding.UTF8.GetString(strm.ToArray()));
-            }
+            var host = SocketsTestHost.CreateWithDefaultEndPoint();
+            var result = await host.ExecuteRequestAsync(path);
+            Assert.Equal(StatusCodes.Status400BadRequest, result.HttpContext.Response.StatusCode);
+            Assert.Equal("Connection ID required", Encoding.UTF8.GetString(result.ResponseBody));
         }
-    }
 
-    public class TestEndPoint : EndPoint
-    {
-        public override Task OnConnectedAsync(Connection connection)
+        [Fact]
+        public async Task CannotUseSendEndpointWithWebSockets()
         {
-            throw new NotImplementedException();
+            var host = SocketsTestHost.CreateWithDefaultEndPoint();
+            var connectionState = host.CreateConnection(transportName: WebSocketsTransport.TransportName);
+            var result = await host.ExecuteRequestAsync("/send", queryString: $"?id={connectionState.Connection.ConnectionId}");
+            Assert.Equal(StatusCodes.Status400BadRequest, result.HttpContext.Response.StatusCode);
+            Assert.Equal("Cannot send to a WebSockets connection", Encoding.UTF8.GetString(result.ResponseBody));
```

Comment:
Could we change this to something like `/send is not supported on WebSocket connections`